### PR TITLE
BREAKING CHANGE: restricting boolean Config types to true and false

### PR DESCRIPTION
**Pull request by Hadis-Fard**
- Added type checking for boolean config option

- ~~Changed the default value of `enableCursorCloseOnCommit` from `undefined` to `false` to match the documents and the rest of boolean options~~

### CODE REVIEW

File: src/connection.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ class Connection extends EventEmitter {
         enableAnsiWarnings: true,
         enableArithAbort: false,
         enableConcatNullYieldsNull: true,
-        enableCursorCloseOnCommit: undefined,
+        enableCursorCloseOnCommit: false,
```
Comment (v-suhame) [inline src/connection.js]: You need to revert the changes made for `enableCursorCloseOnCommit` 😅 More details in #643 & #653

Comment (Hadis-Fard) [reply src/connection.js]: :stuck_out_tongue_closed_eyes: Sure

**Comment by arthurschreiber** [comment]
@Hadis-Fard These changes will be backwards incompatible, because we're changing how non-boolean values are treated.

Could you open a separate pull request that deprecated passing non-boolean values first, similar to #695?

**Comment by Hadis-Fard** [comment]
TODO:
- expand the type errors to the other types as well as boolean types. similar to deprecations in #698
- check for null values in getinitialSql function.
- introducing the changes that are going to be deprecated in #698 

Will resume the work after 698 is merged.

File: src/connection.js [hunk 2]
```diff
@@ -389,6 +366,12 @@ class Connection extends EventEmitter {
     };
   }
 
+  validateBooleanOptions(option, optionName) {
```
Comment (arthurschreiber) [inline src/connection.js]: How do you feel about making this a "regular" function instead of a method on `Connection`? Basically, just like `deprecateNonBooleanConfigValue` on `master`.

**Comment by arthurschreiber** [comment]
Closed in favor of #811.
